src/tests/multihost/docs/docstrings.rst

```diff
@@ -0,0 +1,71 @@
+Docstrings for Test Cases
+=================================
+We use docstrings to document what the test code is testing and what are the expected results. It is a very useful way to describe the test, especially when the test executes on multiple hosts and test code is long and complex. `Python docstrings <https://www.geeksforgeeks.org/python-docstrings>`__ provides more details.
+
+| Besides, these benefits it is also used to automatically add/modify the test cases in an internal test case management system. We use `Betelgeuse <https://betelgeuse.readthedocs.io/en/stable/>`__ for this purpose.
+
+=========================
+Docstrings in SSSD tests
+=========================
+Format explained below is only for the actual test cases written in pytest framework. This format is standardized for better readability and to enable `Betelgeuse <https://betelgeuse.readthedocs.io/en/stable/>`__ scripts to add/modify the tests into the internal test case management system
+
+
+New pytest test file
+--------------------
+| The first line of the test file should be short description of the tests  covered and next line should be left blank
+eg. `test_kcm.py <../alltests/test_kcm.py#L1>`__
+
+Docstring params
+----------------
+Many params are common for all the tests contained in the file. Such params can be mentioned in the start of the file and would apply to each test when the tests are added to the test case management system. If a param for a test is different, we can override it by adding it in the individual test case.
+
+Common Fixed params
+*******************
+These params with their respective values should always be part of test files
+
+**:upstream: yes** This means that the code is being added in github SSSD project
+
+**:casecomponent: sssd** The component package to be tested
+
+**:subsystemteam: sst_idm_sssd** A custom param to categorize the test for a specific team. (internal purpose)
+
```

Comment:
Can we add the following params also,
:caselevel: Acceptance
:caseautomation: Automated
:caseimportance: Medium
:testtype: Functional

Reply:
done